ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Impl/Common.hs

```diff
@@ -350,6 +363,43 @@ data RevalidateTxsResult blk =
    , removedTxs       :: ![Invalidated blk]
    }
 
+-- | Compute snapshot is largely the same as revalidate the transactions
+-- but we ignore the diffs.
+computeSnapshot
+  :: (LedgerSupportsMempool blk, HasTxId (GenTx blk))
+  => MempoolCapacityBytesOverride
+  -> LedgerConfig blk
+  -> SlotNo
+  -> TickedLedgerState blk DiffMK
+     -- ^ The ticked ledger state againt which txs will be revalidated
+  -> LedgerTables (LedgerState blk) ValuesMK
+     -- ^ The tables with all the inputs for the transactions
+  -> TicketNo -- ^ 'isLastTicketNo' and 'vrLastTicketNo'
+  -> [TxTicket (TxMeasure blk) (Validated (GenTx blk))]
+  -> MempoolSnapshot blk
+computeSnapshot capacityOverride cfg slot st values lastTicketNo txTickets =
+  let theTxs = map wrap txTickets
+      wrap = (\(TxTicket tx tk tz) -> (tx, (tk, tz)))
+      unwrap = (\(tx, (tk, tz)) -> TxTicket tx tk tz)
+      ReapplyTxsResult _ val st' =
+        reapplyTxs IgnoreDiffs cfg slot theTxs
+        $ applyDiffForKeysOnTables
+              values
+              (Foldable.foldMap' (getTransactionKeySets . txForgetValidated . fst) theTxs)
+              st
+
+  in snapshotFromIS $ IS {
+         isTxs          = TxSeq.fromList $ map unwrap val
+       , isTxIds        = Set.fromList $ map (txId . txForgetValidated . fst) val
+       , isTxKeys       = emptyLedgerTables
```

Comment:
I think a comment could help explain why `isTxKeys` and `isTxValues` can soundly be empty here.